Rename account_passwords_pam_faillock_audit

marcusburghardt opened the pull request:
#### Description:

All other pam_faillock.so parameters related rules follow a different standard.
It was basically renamed `account_passwords_pam_faillock_audit` to `accounts_passwords_pam_faillock_audit`.

#### Rationale:

Better sooner than later.

marcusburghardt commented:
I will fix the `stable-profiles` test result soon.

marcusburghardt commented:
> @marcusburghardt the rename of a rule will break tailoring for people and will cause troubles for insights

@jan-cerny , this rule was introduced about just two months ago (2022-06-21). Unfortunately it was merged with this misalignment with other similar rules but since it is pretty fresh, I think the risk of breaking eventual tailoring made this meantime is low and acceptable in favor of keeping the project more organized. If we don't fix it now, it would be much painful to fix in the future, which probably wouldn't happen.

Could you give me more details would the impact on insights, please? So I can investigate and recalculate the risk.
If we see the risk is too high, one option is to also keep the former name with an explicit deprecation note.

jan-cerny commented:
The problem is that the Insights generate Ansible Playbooks for rules based on the latest (downstream?) release and if that changes they have a mismatch between set of available rules and set of generated Playbooks.

jan-cerny commented:
@marcusburghardt I believe that the action plan here would be:
* We avoid hard problems by not removing anything from DS whenever we want to rename / split a rule.
* New and old rules should have “superseeds” and “obsoleted by” warnings for the transitional period.
* We keep the “old” rule in the DS for some time, so it is the tailoring that will get screwed.
* It is imperative that we document those “breaking” changes in release notes, ideally Insights should be able to consume them.
* Changes like PR #9462 are worth doing because it improves consistency.

Am I correct?

marcusburghardt commented:
> @marcusburghardt I believe that the action plan here would be:
> 
> * We avoid hard problems by not removing anything from DS whenever we want to rename / split a rule.
> * New and old rules should have “superseeds” and “obsoleted by” warnings for the transitional period.
> * We keep the “old” rule in the DS for some time, so it is the tailoring that will get screwed.
> * It is imperative that we document those “breaking” changes in release notes, ideally Insights should be able to consume them.
> * Changes like PR [Rename account_passwords_pam_faillock_audit #9462](https://github.com/ComplianceAsCode/content/pull/9462) are worth doing because it improves consistency.
> 
> Am I correct?

Yes, that is correct @jan-cerny . I plan to update these PRs and documentation soon.

marcusburghardt commented:
Hi, I plan to resume the work on this PR along the week.

marcusburghardt commented:
I rebased the PR and updated it:

- Created a new rule called `accounts_passwords_pam_faillock_dir` based on the existing `account_passwords_pam_faillock_dir` rule instead of renaming the existing rule
- Included a deprecation warning in the `account_passwords_pam_faillock_dir` rule
- Updated the OVAL ids in the `account_passwords_pam_faillock_dir rule` in order avoid ids duplication

Although the commits ids were changed in favor of better organization after the changes, their contents were not modified beyond the changes described above.

marcusburghardt commented:
Included `highlight` label in this PR to make sure the changes are properly announced in the release notes.

jan-cerny reviewed (changes requested):
@marcusburghardt Thanks for the update. It looks great!  Could you please rebase once more because the CentOS stream 9 Testing farm CI job is now fixed so they switched it to required.

marcusburghardt commented:
> @marcusburghardt Thanks for the update. It looks great! Could you please rebase once more because the CentOS stream 9 Testing farm CI job is now fixed so they switched it to required.

Thanks @jan-cerny . Sure, I will rebase it within the next 2 hours.

marcusburghardt commented:
> > @marcusburghardt Thanks for the update. It looks great! Could you please rebase once more because the CentOS stream 9 Testing farm CI job is now fixed so they switched it to required.
> 
> Thanks @jan-cerny . Sure, I will rebase it within the next 2 hours.

Done.
